qiita_db/sql_connection.py

```diff
@@ -22,57 +22,139 @@
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchlast()
+>>> res
+42
+
+Getting the results of the last SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchindex()
+>>> res
+[[42]]
+
+Getting the results of the specified SQL query
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 43")
+...     res = TRN.execute_fetchindex(0)
+>>> res
+[[42]]
+
+Getting the results of the last SQL query flattened
+>>> with TRN:
+...     TRN.add("SELECT 42, 43, 44")
+...     res = TRN.execute_fetchflatten()
+>>> res
+[42, 43, 44]
+
+Getting the results of the specified SQL query flattened
+>>> with TRN:
+...     TRN.add("SELECT 42, 43, 44")
+...     TRN.add("SELECT 42")
+...     res = TRN.execute_fetchflatten(0)
+>>> res
+[42, 43, 44]
+
+* Transactions
+
+Transaction blocks are created through the same `TRN` variable exported in this
+module. You can add as many SQL commands as you want and execute all of them at
+once, and it will return the results of all the SQL commands. `TRN` should be
+used as a context so manager, and it autocommits the transaction once the last
+context is exited, as long as no error was generated inside the context, in
+which case a rollback is executed.
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 43")
+...     res = TRN.execute()  # The transaction is not committed here
+>>> # The transaction committed here
+>>> res
+[[[42]], [[43]]]
+
+You can have nested transactions and they will not commit until the first
+transaction (represented by entering to the context) is exited:
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     with TRN:
+...         TRN.add("SELECT 43")
+...         res = TRN.execute()
+...     # The transaction is still not committed
+...     TRN.add("SELECT 44")
+...     res = TRN.execute()
+>>> # The transactions committed here
+>>> res
+[[[42]], [[43]], [[44]]]
+
+You can also use results from a previous command in the transaction in a later
+command without partially executing the transaction by using placeholders. The
+placeholders are of the form {int:int:int}, where the first integer is the
+index of the query in the transaction, the second integer is the row in the
+query results where the value exists, and the last integer is the actual index
+of the desired value in the result row.
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 42 + %s", ["{0:0:0}"])
+...     res = TRN.execute()
+>>> res
+[[[42]], [[84]]]
+
+>>> with TRN:
+...     sql = ("SELECT param_set_name, preprocessed_params_id "
+...            "FROM qiita.preprocessed_sequence_454_params "
+...            "WHERE barcode_type = %s ORDER BY preprocessed_params_id")
+...     TRN.add(sql, ["golay_12"])
+...     sql = ("SELECT * FROM qiita.preprocessed_data "
+...            "WHERE preprocessed_params_table = %s "
+...            "AND preprocessed_params_id = %s")
+...     TRN.add(sql, ["preprocessed_sequence_454_params", "{0:0:1}"])
```

Comment:
The indexing here is not intuitive since the a `SELECT *` is being used. Possible to change the `SELECT *` to project only a specific column or columns?